File: mvn-advent-of-code/src/test/java/advent/twenty_seventeen/Day07RecursiveCircusTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day07RecursiveCircusTest {
    @Test
    public void checkReal() {
        DailyProblem<String, Integer> dp = new Day07RecursiveCircus("/twenty_seventeen/Day07-RecursiveCircus-input.txt");

        assertEquals("qibuqqg", dp.getPart1Answer());
        assertEquals(1079, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<String, Integer> dp = new Day07RecursiveCircus("/twenty_seventeen/Day07-RecursiveCircus-example#1.txt");

        assertEquals("tknk", dp.getPart1Answer());
        assertEquals(60, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day07RecursiveCircusTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day07RecursiveCircusTest  extends FlatSpec with Matchers {
  "2017 Day 07 - Input" should "calculate answers" in {
    val dp = new Day07RecursiveCircus("/twenty_seventeen/Day07-RecursiveCircus-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day07RecursiveCircus("/twenty_seventeen/Day07-RecursiveCircus-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
